graphene/src/document.rs

```diff
@@ -50,6 +54,14 @@ impl Document {
 		val.unwrap()
 	}
 
+	pub fn view_mode(&self) -> ViewMode {
+		self.view_mode
+	}
+
+	pub fn update_view_mode(&mut self, mode: ViewMode) {
+		self.view_mode = mode;
+	}
+
```

Comment:
Mutations of the document should usually not be done through function calls but rather Operations but I think it is fine in this case because the contents of the document don't really change